packages/mobile/src/account/profileInfo.ts

```diff
@@ -0,0 +1,326 @@
+// See https://github.com/facebook/react-native/blob/f96478778cc00da8c11da17f9591dbdf928e7437/Libraries/Blob/FileReader.js#L85
+async function responseBuffer(response: Response) {
+  const blob = await response.blob()
+  return blobToBuffer(blob)
+}
+
+// Hacky way to get Buffer from Blob
+// Note: this is gonna transfer the whole data over the RN bridge (as base64 encoded string)
+// and should be avoided for large files!
+function blobToBuffer(blob: Blob) {
+  const reader = new FileReader()
+  reader.readAsDataURL(blob)
+  return new Promise<Buffer>((resolve, reject) => {
+    reader.onerror = () => {
+      reject(reader.error)
+    }
+    reader.onloadend = () => {
+      const result = reader.result
+      if (typeof result !== 'string') {
+        reject(new Error('Unexpected result type'))
+      } else {
+        // Result looks like "data:application/octet-stream;base64,BLMHTkM..."
+        // Extract the base64 part
+        const base64 = result.substr(result.lastIndexOf(',') + 1)
+        resolve(Buffer.from(base64, 'base64'))
+      }
+    }
+  })
+}
+
+class UploadServiceDataWrapper implements OffchainDataWrapper {
+  signer: Address
+  self: Address
+
+  constructor(readonly kit: ContractKit, address: Address) {
+    this.signer = this.self = address
+  }
+
+  async writeDataTo(
+    data: Buffer,
+    signature: Buffer,
+    dataPath: string
+  ): Promise<OffchainErrors | void> {
+    const dataPayloads = [data, signature]
+    const signedUrlsPayload = [
+      {
+        path: dataPath,
+      },
+      {
+        path: `${dataPath}.signature`,
+      },
+    ]
+
+    const hexPayload = ensureLeading0x(
+      Buffer.from(JSON.stringify(signedUrlsPayload)).toString('hex')
+    )
+    const authorization = await this.kit.getWallet().signPersonalMessage(this.signer, hexPayload)
+    const signedUrls = await authorizeURLs(signedUrlsPayload, authorization)
+    await Promise.all(
+      signedUrls.map(({ url, fields }, i) => {
+        const formData = new FormData()
+        for (const name of Object.keys(fields)) {
+          formData.append(name, fields[name])
+        }
+        formData.append('file', dataPayloads[i])
+
+        return fetch(url, {
+          method: 'POST',
+          headers: formData.getHeaders(),
+          // Use getBuffer() which ends up transferring the body as base64 data over the RN bridge
+          // because RN doesn't support Buffer inside FormData
+          // See https://github.com/facebook/react-native/blob/b26a9549ce2dffd1d0073ae13502830459051c27/Libraries/Network/convertRequestBody.js#L34
+          // and https://github.com/facebook/react-native/blob/b26a9549ce2dffd1d0073ae13502830459051c27/Libraries/Network/FormData.js
+          body: formData.getBuffer(),
+        }).then((x) => {
+          if (!x.ok) {
+            Logger.error(TAG + '@writeDataTo', 'Error uploading ' + x.headers.get('location'))
+            throw Error(`Error uploading CIP8 data, with status ${x.status}`)
```

Comment:
Instead of throwing, we should return an `OffChainErrors` object here, as specified in the return type. (In generally, we are trying to eliminate `throw` wherever possible.)